### ISSUE TITLE
IE11 minicart not updating on configurable product page (ES6)

### ISSUE THREAD

**Issue by tommi-martin** [magento/magento2#13820]
### Preconditions

* Magento 2.2+ or
* Any M2 version with Magento_ConfigurableProduct/view/frontend/web/js/options-updater.js present
* Win 8.1 / Internet Explorer 11

### Steps to reproduce

1. Navigate to product configurable product page.
2. Select swatch attribute values
3. Add product to cart

### Expected result

Product is added to cart, minicart updates to display the product in cart.

### Actual result

Product is added to cart, Minicart displays “You have no items in your shopping cart."

### Notes:
The issue is caused by options-updater.js line 26, 

```
changedProductOptions = data.items.find(function (item) {
    return item['product_id'] === productId;
});
```

data.items is an array. 
IE11 does not support ES6 array.prototype.find() method. 
This method is not polyfilled by es6-collections.js

IE Console:
`Object doesn’t support property or method ‘find'`

Only way to update the minicart is leaving the configurable product page after this point. IE console error persists if you return to the configurable product page.

### Possible fixes:
* Update es6-collections.js to es6-shim.js. This would be preferable since as es6-collections.js is no longer maintained. Include new WeakMap implementation since it is not present in es6-shim, perhaps the implementation from es6-collections would be enough.
* Remove array.prototype.find method from all javascript files and replace with supported code.

**Comment by magento-engcom-team** [comment]
@Pumppa, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by brvm** [comment]
Facing the same problem. When adding configurable product in IE10 and IE11, I get exactly the same problem as OP. 

Temporary fix:
Change .find to .filter

**Comment by RetroModernDev** [comment]
I'm working on it.

**Comment by magento-engcom-team** [comment]
Hi @Pumppa. Thank you for your report. 
The issue has been fixed in magento/magento2#14105 by @Frodigo in 2.2-develop branch
Related commit(s):
 - [2235f3e6301dd91a5dd1bcac7376fc75bc738fe9](https://github.com/magento/magento2/commit/2235f3e6301dd91a5dd1bcac7376fc75bc738fe9)

The fix will be available with the upcoming 2.2.5 release.

**Comment by magento-engcom-team** [comment]
Hi @Pumppa. Thank you for your report. 
The issue has been fixed in magento/magento2#14192 by @mastiuhin-olexandr in 2.3-develop branch
Related commit(s):
 - [6d4065a53ab0a066057e156f48cda8df42589598](https://github.com/magento/magento2/commit/6d4065a53ab0a066057e156f48cda8df42589598)

The fix will be available with the upcoming 2.3.0 release.
